src/modules/utils/BOAFormatter.ts

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+
+    The class that defines the BOAFormatter.
+        BOA is an integer with seven decimal points.
+        It is stored in BigInteger without decimal point.
+        Therefore, 1BOA is stored as 10,000,000.
+
+    Copyright:
+        Copyright (c) 2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import JSBI from "jsbi";
+
+/**
+ * Options used to express the amount in BOA unit.
+ */
+export interface IBOAFormatterOption {
+    /**
+     * Whether the integer part displays a comma every 1000 units
+     */
+    commaOnThousand?: boolean;
+
+    /**
+     * Decimal Digits
+     */
+    numberOfDigit?: number;
+}
+
+/**
+ * A class with conversion capabilities to the amount shown in BOA
+ * units and the types of data stored internally
+ */
+export class BOAFormatter {
```

Comment:
```suggestion
export class BOAUnitFormatter {
```
It understand the meaning more.

Reply:
I think BOAFormat is short and good